mod_reforged/hooks/entity/tactical/actor.nut

```diff
@@ -165,3 +172,62 @@
 	{
 	}
 });
+
+::Reforged.HooksMod.hookTree("scripts/entity/tactical/actor", function(q) {
+	q.onActorKilled = @(__original) function( _actor, _tile, _skill )
+	{
+		local wasOverriding = ::Reforged.Config.XPOverride;
+		::Reforged.Config.XPOverride = true;
+		__original(_actor, _tile, _skill);
+		::Reforged.Config.XPOverride = wasOverriding;
+	}
+
+	q.onDeath = @(__original) function( _killer, _skill, _tile, _fatalityType )
+	{
+		__original(_killer, _skill, _tile, _fatalityType);
+
+		// The following is an override of the XP gain system. We award XP based on ratio of total damage dealt to an entity.
+
+		// - All factions who did damage share the XP gained from this actor's death.
+		// - The XP available to a faction is based on the percentage of total damage done by this faction.
+		// - This is divided into Killer XP and Group XP.
+		// - The brothers who did damage are considered "Killers". They share the Killer XP based on the
+		// percentage of damage done by each bro relative to other bros who did damage.
+		// - The group XP is then shared equally among all bros in the company.
+
+		local bros = ::Tactical.Entities.getInstancesOfFaction(::Const.Faction.Player);
+
+		if (::Const.Faction.Player in this.m.RF_DamageReceived)
+		{
+			local XPavailable = this.getXPValue() * this.m.RF_DamageReceived[::Const.Faction.Player].Total / this.m.RF_DamageReceived.Total;
```

Comment:
I would add a check whether `this.m.RF_DamageReceived.Total == 0`.

Currently it is possible that that variable is 0.0 when a character receives 0 total damage. And if they are then killed off by something other than damage (console command/future alternative deaths?) the game crashes

Reply:
Nevermind. the problem has to be solved elsewhere